openfl-tutorials/experimental/Workflow_FedProx_Keras_MNIST.ipynb

```diff
@@ -0,0 +1,1304 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "14821d97",
+   "metadata": {},
+   "source": [
+    "# Workflow Interface Tutorial: Keras MINST"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "bd059520",
+   "metadata": {},
+   "source": [
+    "This tutorial demonstrates how to train a model using TensorFlow Keras on the MNIST dataset, leveraging the new FedAI Workflow Interface. The Workflow Interface provides a novel way to compose federated learning experiments with OpenFL, enabling researchers to handle non-IID data and perform federated averaging with optimizations like FedProx. Through this tutorial, you will learn how to set up the federated learning environment, define the flow, and execute the training process across multiple collaborators.\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "39c3d86a",
+   "metadata": {},
+   "source": [
+    "# What is it?"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "a7989e72",
+   "metadata": {},
+   "source": [
+    "The workflow interface is a new way of composing federated learning expermients with OpenFL. It was borne through conversations with researchers and existing users who had novel use cases that didn't quite fit the standard horizontal federated learning paradigm. "
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "fc8e35da",
+   "metadata": {},
+   "source": [
+    "# Getting Started"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "4dbb89b6",
+   "metadata": {},
+   "source": [
+    "First we start by installing the necessary dependencies for the workflow interface"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f7f98600",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install tensorflow\n",
```

Comment:
New tensorflow does not support Keras APIs without a legacy flag. Suggest taking a look at [TensorFlow + Keras 2 backwards compatibility](https://keras.io/getting_started/) and modify as necessary.

My recommendation is to fully transition this tutorial to Keras 3 APIs instead of using deprecated Keras 2.

Reply:
If we decide to temporarily stick with Keras 2, we also will need to pin `tensorflow` to a compatible version or install `tf-keras` to use legacy code. 

However, since this tutorial is set to use `LocalRuntime` by default, I'm more inclined to say we should upgrade this and FedProx to Keras 3 API and then let the `Task Runner`/`FederatedRunTime` catch up. Which it should in a reasonable time. 

Upgrading FedProx to Keras3 is not something that would naturally happen by updating the Task Runner, so it would be worth the effort to include it in this PR if @yontyon as the bandwidth and/or is still interested, or otherwise an adjacent PR
 
Open to thoughts and suggestions

Reply:
@kta-intel , I've updated the inline implementation to keras3 API. However, I don't think we should replace the original implementation before upgrading the whole project to keras3 (to avoid confusion and versions mixed in the code source). WDYT?
Please review the keras3 implementation as well

Reply:
@kta-intel , @teoparvanov , I moved the mew implementation to the openfl optimizers package. At the moment, the only usage of the keras based fedprox implementation is in this tutorial so no additional changes are required,

Reply:
Awesome, thanks @yontyon ! As pointed out by @kta-intel , this approach has a positive impact on the overall framework.